sam-build-and-deploy/action.yml

```diff
@@ -0,0 +1,58 @@
+name: SAM Build and Deploy
+description: SAM Build and Deploy
+
+inputs:
+  environment:
+    description: Environment
+    required: true
+  microservice-name:
+    description: Microservice name
+    required: true
+  aws-account-id:
+    description: AWS account id
+    required: true
+  aws-access-key-id:
+    description: AWS access key id
+    required: true
+  aws-secret-access-key:
+    description: AWS secret access key
+    required: true
+  aws-region:
+    description: AWS Region
+    required: true
+  s3-bucket:
+    description: S3 bucket
+    required: false
+    default: doo-infrastructure
+
+runs:
+  using: composite
+  steps:
+    - uses: aws-actions/setup-sam@v1
+    - uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ inputs.aws-access-key-id }}
+        aws-secret-access-key: ${{ inputs.aws-secret-access-key }}
+        aws-region: ${{ inputs.aws-region }}

```

Comment:
This part is better to move out to the workflow. So also you don't need those two additional parameters:

```
inputs.aws-access-key-id
inputs.aws-secret-access-key
```

Reply:
For other shared actions, we don't use `aws-actions/configure-aws-credentials@v1` as well and moved this out to the workflow to reduce the number of parameters and because of the other reasons.